scripts/graphics/agentsview.js

```diff
@@ -53,6 +55,18 @@ define([
                 collection.forEach(spriteRenderer.display);
             }
 
+            function clearCache(x,y) {
+                var tileX = ((x % worldWidth) + worldWidth) % worldWidth;
+                var tileY = ((y % worldHeight) + worldHeight) % worldHeight;
+                var key = tileX + "-" + tileY;
+                delete collection.options.agentsCache[key];
+            }
+
+            game.getWorld().setAgentCallback(function(agent, locationFrom, locationTo) {
```

Comment:
Be careful. When an agent die, locationTo will be null. When a agent is created locationFrom will be null.